spec/controllers/users_controller_spec.rb

```diff
@@ -160,4 +160,15 @@
       expect(user.locale).to eq('es')
     end
   end
+
+  describe '#show' do
+    context 'when user not found' do
+      before {allow(controller).to receive(:current_user).and_return(nil) }
+      it 'redirects to the home page' do
+       @controller = DashboardController.new
+       get :index
```

Comment:
This does not test the behavior of `UsersController#show`. You need to do `get :show` with the required :username parameter, like this:

`get :show, params: { username: 'not a real user' }`

You can verify that it's testing the right thing by commenting out the redirect in UsersController#show, which should cause the test to fail.